src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuter.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using AzureDevOpsPolicyConfigurator.Data;
+using AzureDevOpsPolicyConfigurator.Interfaces;
+using Microsoft.TeamFoundation.Policy.WebApi;
+using Microsoft.TeamFoundation.SourceControl.WebApi;
+
+namespace AzureDevOpsPolicyConfigurator.Logic
+{
+    /// <summary>
+    /// PolicyExecuter class.
+    /// </summary>
+    internal class PolicyExecuter : PolicyExecuterBase
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PolicyExecuter"/> class.
+        /// </summary>
+        /// <param name="serializer">Json serializer</param>
+        /// <param name="reader">File reader</param>
+        /// <param name="connectionProvider">Connection provider</param>
+        public PolicyExecuter(IJsonSerializer serializer, IFileReader reader, IConnectionProvider connectionProvider)
+            : base(serializer, reader, connectionProvider)
+        {
+        }
+
+        /// <summary>
+        /// Creates a policy in TFS.
+        /// </summary>
+        /// <param name="policyClient">Policy client</param>
+        /// <param name="types">Types</param>
+        /// <param name="projectId">Team project id</param>
+        /// <param name="repository">Git repository</param>
+        /// <param name="currentPolicy">Branch policy</param>
+        /// <param name="policy">Policy</param>
+        protected override void CreatePolicy(
+            PolicyHttpClient policyClient,
+            IEnumerable<PolicyType> types,
+            Guid projectId,
+            GitRepository repository,
+            BranchPolicies currentPolicy,
+            Policy policy)
+        {
+            var policyConfiguration = this.GetPolicyConfiguration(types, repository, policy);
+
+            var async = policyClient.CreatePolicyConfigurationAsync(policyConfiguration, projectId);
+
+            Log.Debug(this.Serializer.Serialize(async.Result));
+            Log.Info($"Policy has been created. (Branch: {currentPolicy.Branch}, Repository: {repository.Name}, Type: {policy.Type})");
+        }
+
+        /// <summary>
+        /// Updates a policy in TFS.
+        /// </summary>
+        /// <param name="policyClient">Policy client</param>
+        /// <param name="types">Types</param>
+        /// <param name="projectId">Team project id</param>
+        /// <param name="repository">Git repository</param>
+        /// <param name="currentPolicy">Branch policy</param>
+        /// <param name="policy">Policy</param>
+        /// <param name="tfsPolicy">Tfs policy</param>
```

Comment:
Use AzureDevOps instead of TFS, as on-prem product is now also called Azure DevOps Server